src/main/java/com/bunq/sdk/json/MonetaryAccountReferenceTypeAdapter.java

```diff
@@ -17,8 +17,12 @@
 
   @Override
   public void write(JsonWriter out, MonetaryAccountReference value) throws IOException {
-    if (value == null || value.getPointer() == null) {
+    if (value == null || value.isAllFieldNull()) {
       out.nullValue();
+    } else if (value.getPointer() == null) {
```

Comment:
Wouldn't it be nicer to check if the `value.getLabelMonetaryAccount` is not null?

Reply:
But thats already done above ? `    if (value == null || value.isAllFieldNull()) {` via `isAllFieldNull` ? 🤔

Reply:
The statements that follow is purely to know which one to use as both cant be populated with data.

Reply:
There you check if all fields are null. Here you check if the pointer is null. Wouldn't it be nicer to check:
```
else if (value.getLabelMonetaryAccount() != null) {
```

Because in theory it doesn't mean that if not all fields are null and the pointer is null that the Monetary account label is not null.

Reply:
😮 😁 🤦‍♂️ makes sense.